keanu-project/src/test/java/io/improbable/keanu/e2e/regression/RadonHeirarchicalRegression.java

```diff
@@ -0,0 +1,179 @@
+package io.improbable.keanu.e2e.regression;
+
+import io.improbable.keanu.DeterministicRule;
+import io.improbable.keanu.algorithms.variational.optimizer.gradient.GradientOptimizer;
+import io.improbable.keanu.model.regression.RegressionModel;
+import io.improbable.keanu.model.regression.RegressionRegularization;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.util.csv.ReadCsv;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.HalfGaussianVertex;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class RadonHeirarchicalRegression {
+
+    @Rule
+    public DeterministicRule rule = new DeterministicRule();
+
+    private List<Data> radonData;
+
+    @Before
+    public void readRadonCSV() {
+        radonData = ReadCsv.fromResources("data/datasets/radon/radon.csv")
```

Comment:
Could we put this in `CsvDataResource` and load in the same way as `DiabetesLinearRegression`? Would be more consistent.

Reply:
good point - I want to read in rows (a list of `Data`) whereas `CsvDataResource` only supports vectorised columns at the minute (reading into a tensor). Let's add the support for reading in as rows as a separate PR